src/it/scala/com/exasol/spark/PredicatePushdownSuite.scala

```diff
@@ -51,4 +48,34 @@ class PredicatePushdownSuite extends FunSuite with BaseDockerSuite with DataFram
     assert(result === Set((1, "Berlin"), (2, "Paris")))
   }
 
+  test("date and timestamp should be read correctly") {
+    import java.sql.Date
+
+    createDummyTable()
+    val df = spark.read
+      .format("exasol")
+      .option("host", container.host)
+      .option("port", s"${container.port}")
+      .option("query", s"SELECT date_info, updated_at FROM $EXA_SCHEMA.$EXA_TABLE")
+      .load()
+
+    val resultDate = df.collect().map(_.getDate(0))
+    assert(resultDate.contains(Date.valueOf("2017-12-31")))
+
+    val minTimestamp = Date.valueOf("2017-12-30")
+    val resultTimestamp = df.collect().map(_.getTimestamp(1)).map(x => x.after(minTimestamp))
+    assert(!resultTimestamp.contains(false))
+  }
+
```

Comment:
Could you please also add a read with filter condition for date and timestamp columns?
For example:
```scala
...
.option("query", ...)
.load()
.filter($"date_info" === "2017-12-31")
.filter($"update_at" === "2017-12-31 10:00:00")
```
This will also cover the predicate pushdown changes for these types.

However, it is also okay to add them in a follow up pull request.